odoo/addons/base/views/res_company_views.xml

```diff
@@ -57,14 +67,28 @@
             <field name="name">res.company.tree</field>
             <field name="model">res.company</field>
             <field name="arch" type="xml">
-                <tree string="Companies">
+                <tree string="Companies" decoration-muted="not active">
                     <field name="sequence" widget="handle"/>
                     <field name="name"/>
                     <field name="partner_id"/>
                     <field name="child_ids" widget="many2many_tags" options="{'color_field': 'color'}"/>
+                    <field name="active" invisible="1"/>
                 </tree>
             </field>
         </record>
+        <record id="view_company_branches_tree" model="ir.ui.view">
+            <field name="name">res.company.branches.tree</field>
+            <field name="model">res.company</field>
+            <field name="inherit_id" ref="base.view_company_tree"/>
+            <field name="mode">primary</field>
+            <field name="priority">100</field>
+            <field name="arch" type="xml">
+                <field name="active" position="attributes">
+                    <attribute name="widget">boolean_toggle</attribute>
+                    <attribute name="invisible">0</attribute>
+                </field>
```

Comment:
Do we really need this if we are already in a tree view? One can already use the generic "(Un)Archive" actions

Reply:
It was requested by Benj, so @tsb-odoo your call ;)

Reply:
@tsb-odoo up?

Reply:
If I remember correctly, when I first tested, the archived branches were excluded from the view. So, the user couldn't access the archived branch(es) to unarchive. Re-testing now, given the inactive branches are not filtered out from the list view, the toggle button is no more important.

Reply:
@william-andre @tsb-odoo Alright, I removed the view with the toggle buttons.